examples/vision/yolov3.py

```diff
@@ -0,0 +1,443 @@
+    netout[..., 4:] = _sigmoid(netout[..., 4:])
+    netout[..., 5:] = netout[..., 4][..., np.newaxis] * netout[..., 5:]
+    netout[..., 5:] *= netout[..., 5:] > thresh
+    for i in range(grid_h * grid_w):
+        row, col = i / grid_w, i % grid_w
+        for b in range(nb_box):
+            # 4th element is objectness score
+            objectness = netout[int(row)][int(col)][b][4]
+            # objectness = netout[..., :4]
+            if objectness.all() <= thresh:
+                continue
+            # first 4 elements are x, y, w, and h
+            x, y, w, h = netout[int(row)][int(col)][b][:4]
+            x = (col + x) / grid_w  # center position, unit: image width
+            y = (row + y) / grid_h  # center position, unit: image height
+            w = anchors[2 * b + 0] * np.exp(w) / shape[1]  # unit: image width
+            h = anchors[2 * b + 1] * np.exp(h) / shape[0]  # unit: image height
+            # last elements are class probabilities
+            classes = netout[int(row)][col][b][5:]
+            box = BoundBox(
+                x - w / 2, y - h / 2, x + w / 2, y + h / 2, objectness, classes
+            )
+            boxes.append(box)
+    return boxes
+
+
+def _interval_overlap(interval_a, interval_b):
+    x1, x2 = interval_a
+    x3, x4 = interval_b
+    if x3 < x1:
+        if x4 < x1:
+            return 0
+        else:
+            return min(x2, x4) - x1
+    else:
+        if x2 < x3:
+            return 0
+        else:
+            return min(x2, x4) - x3
+
+
+# sigmoid function
+def _sigmoid(x):
+    return 1.0 / (1.0 + np.exp(-x))
+
+
+# Intersection over Union
+def iou(box1, box2):
+    intersect_w = _interval_overlap([box1.xmin, box1.xmax], [box2.xmin, box2.xmax])
+    intersect_h = _interval_overlap([box1.ymin, box1.ymax], [box2.ymin, box2.ymax])
+    intersect = intersect_w * intersect_h
+    w1, h1 = box1.xmax - box1.xmin, box1.ymax - box1.ymin
+    w2, h2 = box2.xmax - box2.xmin, box2.ymax - box2.ymin
+    union = w1 * h1 + w2 * h2 - intersect
+    return float(intersect) / union
+
+
+# box pos correction
+def correct_yolo_boxes(boxes, image_h, image_w, shape):
+    if (float(shape[1]) / image_w) < (float(shape[0]) / image_h):
+        new_w = shape[1]
+        new_h = (image_h * shape[1]) / image_w
+    else:
+        new_h = shape[1]
+        new_w = (image_w * shape[0]) / image_h
+    for i in range(len(boxes)):
+        x_offset, x_scale = (shape[1] - new_w) / 2.0 / shape[1], float(new_w) / shape[1]
+        y_offset, y_scale = (shape[0] - new_h) / 2.0 / shape[0], float(new_h) / shape[0]
+        boxes[i].xmin = int((boxes[i].xmin - x_offset) / x_scale * image_w)
+        boxes[i].xmax = int((boxes[i].xmax - x_offset) / x_scale * image_w)
+        boxes[i].ymin = int((boxes[i].ymin - y_offset) / y_scale * image_h)
+        boxes[i].ymax = int((boxes[i].ymax - y_offset) / y_scale * image_h)
+
+
+# Non-max suppression (detecting each object once)
+def non_max_suppression(boxes, thresh):

```

Comment:
This is an example of something that is general and useful to demonstrate -- good!